### PR TITLE
fix: harden GroupMe webhook handling and reach full line/function coverage

### PR DESCRIPTION
**Pull request by oddrationale**
## Summary

Comprehensive code-review follow-up to the OpenClaw 2026.6.1 modernization (#60). Tightens behavior, hardens and de-duplicates the test suite, and reaches 100% statement/function/line coverage with an audited branch floor.

## Behavior changes
- **Warn when `callbackToken` is unconfigured** at gateway start — otherwise inbound callbacks run with no token authentication (only group-id binding, which isn't secret).
- **Report `running: false` / `lastStopAt`** when a gateway account stops (previously status stayed "running" after shutdown).
- **Remove the always-true `replay`/`rateLimit` `enabled` resolved flags** and the unreachable branches they guarded; map webhook body-error codes to HTTP status via a lookup table instead of an `if/else` chain.

## Tests & coverage
- Remove `@ts-nocheck` from `channel.test.ts` and type it fully (surfaced/fixed several arg-shape drifts against the real `ChannelPlugin` contract).
- Replace the silent-pass socket guard in `monitor.test.ts` with **visible skips** (it could previously pass with zero assertions when sockets can't bind).
- Extract shared inbound/onboarding test helpers (removed ~250 lines of duplication).
- Add edge-case suites (malformed proxy/host parsing, oversized/timeout body handling, non-`Error` throws, unconfigured status, etc.).
- **Coverage: 100% statements / 100% functions / 100% lines / ~93% branches.** Thresholds floored at `100/100/100/92`. Every uncovered branch was audited; the remainder is legitimate trust-boundary defense, with documented `/* v8 ignore */` on genuinely-unreachable lines.

## Tooling & docs
- Tighten `knip` entry globs so dead exports are actually detected; un-export internal-only symbols.
- Pin Node via `.node-version` (22) and bump `@types/node` to `^22`.
- Add a dated code-review document; correct a stale `CLAUDE.md` env-var note and clarify inbound-vs-outbound media limits in the README.

## Verification
`biome check`, `tsc --noEmit`, unit (252), integration (19), `build`, `knip`, and the coverage gate all pass on Node 22.22.3.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/integration/onboarding-http-boundary.test.ts [hunk 1]
```diff
@@ -1,64 +1,11 @@
-import type { OpenClawConfig, WizardPrompter } from "openclaw/plugin-sdk/core";
+import type { OpenClawConfig } from "openclaw/plugin-sdk/core";
 import { DEFAULT_ACCOUNT_ID } from "openclaw/plugin-sdk/core";
-import type { RuntimeEnv } from "openclaw/plugin-sdk/runtime-env";
-import { afterEach, describe, expect, it, vi } from "vitest";
+import { afterEach, describe, expect, it, type vi } from "vitest";
```
[the remark this answers is not in the document]
Comment (oddrationale) [reply tests/integration/onboarding-http-boundary.test.ts:3]: Verified — this is a false positive, so leaving as-is. `vi` is referenced only inside `typeof vi.fn` (lines 71/75/76), which is a *type query* (a type position), not a runtime value. The type-only import is therefore valid TypeScript, and it's actually required by our Biome `useImportType: "error"` rule — a plain value import fails lint here because `vi` is never used as a value in this file. `tsc --noEmit` and this integration test both pass with the type-only import.

File: .node-version [hunk 2]
```diff
@@ -0,0 +1 @@
+22
```
[the remark this answers is not in the document]
Comment (oddrationale) [reply .node-version:1]: Good general flag, but it's intentional and safe for this repo's toolchain, so leaving as-is. `.node-version` is consumed only by fnm locally, which resolves a bare major to the latest installed 22.x — and with `FNM_RESOLVE_ENGINES` it also honors the `>=22.19.0` engines floor. CI does not read `.node-version`; it uses a hardcoded `actions/setup-node` matrix (`node-version: [22, 24]`), and setup-node's `22` likewise resolves to the latest 22.x. So no consumer here would resolve `22` to an out-of-range `22.0.0`. The bare major deliberately tracks "latest 22.x" and mirrors the CI lane; happy to pin an exact patch later if we prefer reproducibility over freshness.
